sqlglot/dialects/snowflake.py

```diff
@@ -109,6 +109,52 @@ def _datatype_sql(self, expression):
     return self.datatype_sql(expression)
 
 
+def _values_sql(self, expression: exp.Values) -> str:
+    """Due to a bug in Snowflake we want to make sure that all columns in a VALUES table alias are unquoted.
+
+    We also want to make sure that after we find matches where we need to unquote a column that we prevent users
+    from adding quotes to the column by using the `identify` argument when generating the SQL.
+    """
+    alias = expression.args.get("alias")
+    unquoted_a_column = False
+    if alias:
+        for column in alias.args.get("columns", []):
+            column.set("quoted", False)
```

Comment:
delete this line, it's not necessary and we shouldn't be mutating things in the generator

Reply:
I don't understand. If I remove this line then we don't unquote anything.

Reply:
then the code isn’t correct. test it with identify=True

Reply:
Works as expected:
```
>>> sqlglot.parse_one("SELECT `c0`, `c1` FROM (VALUES (1, 2), (3, 4)) AS `t0`(`c0`, `c1`)", read="spark").sql(dialect="snowflake", identify=True)
'SELECT c0, c1 FROM (VALUES (1, 2), (3, 4)) AS "t0"(c0, c1)'
```

Reply:
why would you need to set quoted then?

Reply:
i see. it’s on only inverse true. need a new flag on no identify

Reply:
This line I think is where we are misaligned: https://github.com/tobymao/sqlglot/blob/e69fa6ce02e136193a404dd56bb0d0e86870601f/sqlglot/generator.py#L549

It is saying that if either `identify=True` or the column is defined as `quoted` then we add the quotes. So if we just set `identify` as `False` then we still have `quoted` set to true and therefore the quote are set. These lines where I set `quoted` to `False` mean that we ensure that both `quoted` is `False` and `identify` is `False` and therefore we don't add the quotes on those columns we are targeting.

Reply:
yea but it’s a mutation. we should add a force no identify

Reply:
Well I still want to identify references that aren't the column. I just want to modify the columns that are part of the values alias.

Reply:
right l, so only no identify should be used for those nodes. or you need to do this in a non mutating manner, maybe through transform

Reply:
So no identify on the nodes won't work as we discussed above. I don't see how we would do a forcing solution without mutating them.

Can explain what you mean by using a transform? An example you can reference?

Reply:
What if we copied the original expression instead? I think we already do something similar for `STRING_AGG` to avoid mutations in the generator.

Reply:
right now you’re calling set directly on the nodes. this is a mutation .

instead of you call transform on the select node, you will make a copy and not mutate the original. your original code used a transform i think.

Reply:
> your original code used a transform i think.

Do you mean `STRING_AGG`, or Ryan's? In any case, just to clarify: I meant something like:

```python
# Do this if a mutation is needed
expression = expression.copy()
...
```

Reply:
i mean ryan.

i’m suggesting

expression.transform(…)